src/libertem/corrections/corrset.py

```diff
@@ -1,10 +1,54 @@
 import numpy as np
 import sparse
+import primesieve
 
 from libertem.common import Slice
 from libertem.corrections.detector import correct
 
 
+def factorizations(n, primes):
+    n = np.array(n)
+    debug_n = n
+    factorization = np.zeros((len(n), len(primes)), dtype=n.dtype)
+    if np.max(n) > np.max(primes):
+        raise ValueError("np.max(n) > np.max(primes)")
+    while np.any(n > 1):
+        zero_modulos = (n[:, np.newaxis] % primes[np.newaxis, :]) == 0
+        factorization[zero_modulos] += 1
+        f = np.prod(primes[np.newaxis, :]**zero_modulos, axis=1)
+        n = n // f
+
+    assert np.all(np.prod(primes ** factorization, axis=1) == debug_n)
+    return factorization
+
+
+def min_disjunct_multiplier(excluded, size):
+    '''
+    Calculate a small integer i for which i * n not in "excluded" for any n > 0
+
+    To make sure that the tile shape negotiation retains as much flexibility as possible,
+    it is important to find a small integer and not just any integer that fulfills
+    this condition.
+    '''
+    # This is guaranteed to include "size" for realistic dimension sizes
+    primes = np.array(primesieve.primes(size*2))
```

Comment:
This could directly use `primesieve.numpy.primes`, which is supposed to be [three times faster](https://github.com/kimwalisch/primesieve-python#numpy-support)

Reply:
Thx! :-)